src/SymfonyCache/KernelClient.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use Http\Client\HttpAsyncClient;
+use Http\Promise\FulfilledPromise;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory;
+use Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Zend\Diactoros\ServerRequest;
+
+/**
+ * An implementation of HttpAsyncClient that allows direct routing through the
+ * Symfony HttpCache kernel without executing a real HTTP request.
+ * It uses the HttpFoundationFactory and the Zend DiactorosFactory to convert
+ * between PSR-7 requests and responses. Both are optional dependencies of this
+ * package thus existence of the respective classes is checked for in the
+ * constructor of this client. It is only needed if you have a single node
```

Comment:
It is only needed if => It can only be used if

and i would put this information above the details about converting requests. (ah and that will need to change anyways when we use php-http instead)